### PR TITLE
Ported `run-with-mono.sh` bash script to Objective C version

### PR DESCRIPTION
**Pull request by kenkendk**
This now works better with MacOS Catalina security features

### CODE REVIEW

File: Installer/OSX/launchers/duplicati-cli.m [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#import "run-with-mono.h"
+
+NSString * const ASSEMBLY = @"Duplicati.CommandLine.exe";
+NSString * const APP_NAME = @"Duplicati.CommandLine";
+int const MONO_VERSION_MAJOR = 4;
```
Comment (warwickmm) [inline Installer/OSX/launchers/duplicati-cli.m]: I think the changes in pull request #3844 that updated the target framework to 4.6.2 require a minimum mono version of  5.0.0 (see comment [here](https://github.com/duplicati/duplicati/pull/3844#issuecomment-525982565)).  Does this value need to be consistent with that?

Comment (kenkendk) [reply Installer/OSX/launchers/duplicati-cli.m]: Yes, I will update it to 5.0.

**Comment by kenkendk** [comment]
The solution does not work for the commandline. The [NSTask documentation](https://developer.apple.com/documentation/foundation/nstask/1407627-standardoutput?language=objc) states that it should forward the output to the parent, but I see no output when invoking from the commandline.

**Comment by warwickmm** [comment]
I don't have a Mac, so I can't provide much feedback here.  @kenkendk, please merge if you think this is good to go.

Also, do you think this addresses issue #3957?

**Comment by kenkendk** [comment]
> Also, do you think this addresses issue #3957?

Yes. This fixes it.

**Pull request merged**